### PR TITLE
Add cli args

### PR DESCRIPTION
**Pull request by kahuang**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -26,15 +29,26 @@ func (f Files) Less(i, j int) bool { return f[i].order < f[j].order }
 
 const SSIZE_MAX = 9223372036854775807
 
+type CLIOpts struct {
+	Root            string `long:"root" short:"r" description:"root path to walk from" required:"true"`
+	EnableHibernate bool   `long:"enable-hibernate" description:"sleep forever after finishing, useful if this is a sidecare container"`
```
Comment (jacksontj) [inline main.go]: sidecare -> sidecar

Comment (jacksontj) [inline main.go]: Maybe `ExitOnFinish` ?

File: main.go [hunk 2]
```diff
@@ -81,6 +95,11 @@ func main() {
 		go worker(wg, ch)
 	}
 	wg.Wait()
+	if opts.EnableHibernate {
+		for {
+			time.Sleep(time.Hour)
```
Comment (jacksontj) [inline main.go:98]: You can actually just create a channel and block on it IIRC (to avoid the wake-up every hour ;) )

**Review by tvi — COMMENTED** [review]
please update vendor directory as well

**Review by jacksontj — APPROVED** [review]

**Comment by kahuang** [comment]
> please update vendor directory as well

Long term we'll fork/cache these somewhere, as the current reasoning for checking in vendor is repeatable builds and concerns around these libraries disappearing or malicious actors having control over the repo

**Pull request merged**